File: MedLabO UI/medlabo_desktop/lib/models/termin/termin.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:medlabo_desktop/models/pacijent/pacijent.dart';
import 'package:medlabo_desktop/models/racun/racun.dart';
import 'package:medlabo_desktop/models/termin_test/termin_test.dart';
import 'package:medlabo_desktop/models/uposlenik/medicinsko_osoblje.dart';
import 'package:medlabo_desktop/models/usluga/usluga.dart';
import 'package:medlabo_desktop/models/zakljucak/zakljucak.dart';

part 'termin.g.dart';

@JsonSerializable()
class Termin {
  String? terminID;
  String? dtTermina;
  bool? status;
  String? napomena;
  String? odgovor;
  bool? obavljen;
  bool? rezultatTermina;
  String? rezultatTerminaPDF;
  List<Usluga>? terminUsluge;
  List<TerminTest>? terminTestovi;
  bool? isDeleted;
  Pacijent? pacijent;
  MedicinskoOsoblje? medicinskoOsoblje;
  Racun? racun;
  Zakljucak? zakljucak;

  Termin(
      {this.terminID,
      this.dtTermina,
      this.status,
      this.napomena,
      this.odgovor,
      this.obavljen,
      this.rezultatTermina,
      this.rezultatTerminaPDF,
      this.terminUsluge,
      this.terminTestovi,
      this.isDeleted,
      this.pacijent,
      this.medicinskoOsoblje,
      this.racun,
      this.zakljucak});

  factory Termin.fromJson(Map<String, dynamic> json) => _$TerminFromJson(json);

  Map<String, dynamic> toJson() => _$TerminToJson(this);
}

File: MedLabO.Models/Termin/Termin.cs
﻿using System.ComponentModel.DataAnnotations;
using MedLabO.Models.Pacijent;

namespace MedLabO.Models.Termin
{
    public class Termin : SoftDeleteEntity
    {
        public Guid TerminID { get; set; }
        public DateTime DTTermina { get; set; }
        public bool? Status { get; set; }
        [MaxLength(300, ErrorMessage = "Maksimalna dužina napomene je 300 karaktera.")]
        public string? Napomena { get; set; }
        [MaxLength(300, ErrorMessage = "Maksimalna dužina odgovora je 300 karaktera.")]
        public string? Odgovor { get; set; }
        [MaxLength(300, ErrorMessage = "Maksimalna dužina razloga otkazivanja je 300 karaktera.")]
        public string? RazlogOtkazivanja { get; set; }
        public bool Obavljen { get; set; } = false;
        public bool RezultatDodan { get; set; }
        public bool ZakljucakDodan { get; set; }
        public bool Placeno { get; set; }
        public byte[]? RezultatTerminaPDF { get; set; }

        public ICollection<Usluga.Usluga> TerminUsluge { get; set; } = new List<Usluga.Usluga>();
        public ICollection<TerminTest> TerminTestovi { get; set; } = new List<TerminTest>();

        public string? PacijentID { get; set; }
        public PacijentWithoutTermini? Pacijent { get; set; }

        public string? MedicinskoOsobljeID { get; set; }
        public MedicinskoOsoblje? MedicinskoOsoblje { get; set; }

        public string? RacunID { get; set; }
        public Racun? Racun { get; set; }

        public string? ZakljucakID { get; set; }
        public Zakljucak? Zakljucak { get; set; }
    }
}
